sdk/ai/azopenaiassistants/models_custom.go

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azopenaiassistants
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+// OpenAI's reference doc for Assistants: https://platform.openai.com/docs/api-reference/assistants
+
+// MessageAttachmentToolDefinition specifies the tools the files being attached will be
+// associated with.
+type MessageAttachmentToolDefinition struct {
+	// CodeInterpreterToolDefinition should be set if you attach files for the CodeInterpreter tool.
+	CodeInterpreterToolDefinition *CodeInterpreterToolDefinition
+
+	// FileSearchToolDefinition should be set if you attach files for the FileSearchToolDefinition tool.
+	FileSearchToolDefinition *FileSearchToolDefinition
+}
+
+// MarshalJSON implements the json.Marshaller interface for type MessageAttachmentToolDefinition.
+func (m *MessageAttachmentToolDefinition) MarshalJSON() ([]byte, error) {
+	switch {
+	case m.CodeInterpreterToolDefinition != nil:
```

Comment:
This won't handle the case when sending a JSON `null` (unclear if that's important). See the `populate()` func that gets generated in the `models_serde.go` file.

Reply:
Fixed with a little if/else so we still call json.Marshal(). The service does validation on the value, so there's no reason for me to.